src/HotChocolate/Language/src/Language.Rewriters/DefaultSyntaxNavigator.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+using static HotChocolate.Language.Rewriters.LangRewritersResources;
+
+namespace HotChocolate.Language.Rewriters;
+
+/// <summary>
+/// Represents the default implementation of <see cref="ISyntaxNavigator" />
+/// </summary>
+public class DefaultSyntaxNavigator : ISyntaxNavigator
+{
+    private readonly List<ISyntaxNode> _ancestors = new();
+
+    /// <inheritdoc cref="ISyntaxNavigator.Push"/>
+    public void Push(ISyntaxNode node) => _ancestors.Add(node);
+
+    /// <inheritdoc cref="ISyntaxNavigator.Pop"/>
+    public void Pop(out ISyntaxNode node)
+    {
+        if (_ancestors.Count == 0)
+        {
+            throw new InvalidOperationException(DefaultSyntaxNavigator_NoAncestors);
```

Comment:
Lets change this implementation. 

Lets introduce:
`ISyntaxNode Pop()`
and
`public bool TryPop(out ISyntaxNode node)`


Pop will throw like you do now and trypop allows the user to use it safely